File: ecodes_mobile/lib/model/rating.dart
import './product.dart';
import './seller.dart';
import './user.dart';
import 'package:json_annotation/json_annotation.dart';



part 'rating.g.dart';

@JsonSerializable(explicitToJson: true)
class Rating {
  int? productId;
  int? buyerId;
  int? sellerId;
  DateTime? date;
  String? description; 
  double? mark;
  User? buyer;
  Product? product;
  Seller? seller;

  Rating(){

  }

 factory Rating.fromJson(Map<String, dynamic> json) => _$RatingFromJson(json);

  /// Connect the generated [_$RatingToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$RatingToJson(this);


}

File: eCodes.Services/Database/Rating.cs
using System;
using System.Collections.Generic;

namespace eCodes.Services.Database
{
    public partial class Rating
    {
        public int RatingId { get; set; }
        public int SellerId { get; set; }
        public int BuyerId { get; set; }
        public DateTime Date { get; set; }
        public string Description { get; set; } = null!;
        public decimal Mark { get; set; }
        public int ProductId { get; set; }

        public virtual Buyer Buyer { get; set; } = null!;
        public virtual Product Product { get; set; } = null!;
        public virtual Seller Seller { get; set; } = null!;
    }
}
